### DISCUSSION TITLE
AirVPN & Wireguard

### DISCUSSION THREAD

**Discussion by hmdrmch** [qdm12/gluetun#1492] [Q&A] [answered]
Hi,

I'm hoping someone can help me.
I've set up the gluetun service with AirVPN as provider using Wireguard.
It connects fine, but everytime the service is restarted it connects to a new server.

I've entered the private key, preshared key, WG addresses and relevant port in the config from the Wireguard conf - which I've generated from the AirVPN site and have chosen a single server.

Does anyone have an idea as to why it connects to a new server upon reboot? I would like to connect to the same server each time, so I have the same public IP each time.

**Comment by hmdrmch** [comment]
I think I found a workaround.

If the container is set up to be using custom instead of airvpn as provider, it connects to the same server upon reboot.

**Comment by qdm12** [comment, marked as answer]
Use `VPN_ENDPOINT_IP` with the endpoint VPN server IP you want. It's written in https://github.com/qdm12/gluetun/wiki/OpenVPN-options which is a bit obscure for Wireguard since it's a variable common to both openvpn and wireguard in Gluetun. (EDIT: split it in https://github.com/qdm12/gluetun/wiki/VPN-options)

Alternatively you can also narrow down server filters, see https://github.com/qdm12/gluetun/wiki/AirVPN#optional-environment-variables for example setting a particular server hostname with `SERVER_HOSTNAMES` will likely re-use the same endpoint IP and you *should* always get the same public ip address.
